### PR TITLE
Update onnx

### PR DESCRIPTION
**Pull request by askhade**
**Description**: 
Update onnx to the latest in master (in preparation to support onnx 1.10)
- Disable tests for unimplemented ops
- Updates to function body initialization in accordance with the FunctionProto changes for IRv8

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/graph/contrib_ops/onnx_function_util.cc [hunk 1]
```diff
@@ -102,8 +102,61 @@ bool BuildFunctionProto(FunctionProto& functionProto, const OpSchema& schema,
                         const std::vector<FunctionBodyHelper::NodeDef>& node_defs,
                         const std::vector<OperatorSetIdProto>& relied_opsets) {
   BuildNodes(functionProto, node_defs);
-  schema.BuildFunction(functionProto, relied_opsets);
+  for (auto& relied_opset : relied_opsets) {
```
Comment (gramalingam) [inline onnxruntime/core/graph/contrib_ops/onnx_function_util.cc:105]: Now that this function has been moved to onnx, can't we delete this here (and call the onnx version)?

Comment (gramalingam) [reply onnxruntime/core/graph/contrib_ops/onnx_function_util.cc:105]: Delete previous function too? And call the one in FunctionBodyHelper?

File: onnxruntime/core/graph/function.cc [hunk 2]
```diff
@@ -177,22 +194,12 @@ static std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const Graph& graph
...
+static std::unordered_map<std::string, int> GetFunctionOpsetImports(const ONNX_NAMESPACE::FunctionProto& func_proto, const std::unordered_map<std::string, int>& graph_imports) {
+  std::unordered_map<std::string, int> function_opset_imports{graph_imports};
   for (const auto& opset_import : func_proto.opset_import()) {
-    auto opset_version = static_cast<int>(opset_import.version());
-    auto result = function_opset_imports.insert({opset_import.domain(), opset_version});
-    ORT_ENFORCE((result.first->second == opset_version),
-                "ONNX model does not support multiple opset versions for a domain. Model imports opset version ",
-                result.first->second, " for domain ", result.first->first, " and function is trying to import opset version ",
-                opset_version, " for the same domain");
+    function_opset_imports.insert({opset_import.domain(), static_cast<int>(opset_import.version())});
```
Comment (gramalingam) [inline onnxruntime/core/graph/function.cc:201]: May be worth adding a line of comment (and pointer to the other compatibility check): is it a precondition that the graph-imports and function-imports don't conflict at this point?

File: onnxruntime/test/onnx/main.cc [hunk 3]
```diff
@@ -583,11 +583,34 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
...
       {"loop13_seq", "ORT api does not currently support creating empty sequences (needed for this test)", {}},
+      {"cast_FLOAT_to_BFLOAT16", "onnx generate bfloat tensor as uint16 type", {}},
+      {"cast_BFLOAT16_to_FLOAT", "onnx generate bfloat tensor as uint16 type", {}},
+      {"castlike_FLOAT_to_BFLOAT16", "Depends on cast.", {}},
+      {"castlike_BFLOAT16_to_FLOAT", "Depends on cast", {}},
+      {"castlike_FLOAT_to_BFLOAT16_expanded", "Depends on cast.", {}},
+      {"castlike_BFLOAT16_to_FLOAT_expanded", "Depends on cast", {}},
+      {"castlike_FLOAT_to_STRING", "Numpy float to string has unexpected rounding for some results.", {}},
+      {"castlike_FLOAT_to_STRING_expanded", "Numpy float to string has unexpected rounding for some results.", {}},
+      {"bernoulli", "By design. Test data is for informational purpose becasue the generator is non deterministic."},
```
Comment (gramalingam) [inline onnxruntime/test/onnx/main.cc:597]: nit: becasue => because

File: onnxruntime/core/graph/contrib_ops/onnx_function_util.cc [hunk 4]
```diff
@@ -102,8 +102,61 @@ bool BuildFunctionProto(FunctionProto& functionProto, const OpSchema& schema,
...
+    auto func_opset_version = get_version(node.domain(), func_imports);
+    auto model_opset_version = get_version(node.domain(), graph_imports);
+
+    if (func_opset_version == -1 && model_opset_version == -1) {
+      // No opset import for domain in graph and function imports.
+      return false;
+    } else if (func_opset_version == model_opset_version || model_opset_version == -1 || func_opset_version == -1) {
+      // If the opset versions are same or opset import for node.domain is present in one of the imports return true.
+      // ONNX allows model_opset_version == -1 but not func_opset_version == -1
+      // We relax the check in ORT to support pre-existing models.
```
Comment (gramalingam) [inline onnxruntime/core/graph/contrib_ops/onnx_function_util.cc:139]: This discrepancy should eventually be resolved, right? If so, may be worth adding a TODO comment to that effect.

File: onnxruntime/core/graph/function.cc [hunk 5]
```diff
@@ -282,11 +289,17 @@ FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
   // as we might make some modifications to the FunctionProto along the way
 
   const auto* node_in_parent_graph = parent_graph_->GetNode(node_index);
+  // For schema defined functions get the version from the node in parent graph.
+  // For the functions which do not have schema defined (model local functions) 
+  // get the since version from the version in opset imports using the domain.
+  auto since_version = node_in_parent_graph->SinceVersion() == -1 
+      ? GetVersionForDomain(node_in_parent_graph->Domain(), body_.MainGraph().DomainToVersionMap())
```
Comment (gramalingam) [inline onnxruntime/core/graph/function.cc:296]: Seems a bit unintuitive. Why is this needed?

**Comment by gramalingam** [comment]
Is this supposed to handle model-local functions? I don't see a test for that. I don't also see where in the graph construction the resolution of a node's target to a model-local function happens. Is that future work?

File: onnxruntime/core/graph/function.cc (hunk 2 above)
Comment (askhade) [reply onnxruntime/core/graph/function.cc:201]: Nope it is not.... We simply use the opset imports provided as part of function proto... If ORT does not have a specialized kernel for this function op then during function inlining any discrepancy will be caught.

File: onnxruntime/core/graph/contrib_ops/onnx_function_util.cc (hunk 4 above)
Comment (askhade) [reply onnxruntime/core/graph/contrib_ops/onnx_function_util.cc:139]: as per offline discussion removing this function altogether.

File: onnxruntime/core/graph/function.cc (hunk 5 above)
Comment (askhade) [reply onnxruntime/core/graph/function.cc:296]: For schema defined functions we use the since version from schema and for model local functions we get the since version from opset imports.

**Review by gramalingam — DISMISSED** [review]

**Review by snnn — APPROVED** [review]

**Comment by yan12125** [comment]
Maybe ONNX protobuf files also need updates? https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/core/protobuf

**Comment by snnn** [comment]
@yan12125 These files seems not used. Let me see if I can delete them.

**Review by snnn — APPROVED** [review]
